src/main/java/com/lineate/testyourlexicon/achievements/AchievementManager.java

```diff
@@ -0,0 +1,45 @@
+package com.lineate.testyourlexicon.achievements;
+
+import com.lineate.testyourlexicon.entities.AchievementEntity;
+import com.lineate.testyourlexicon.entities.User;
+import com.lineate.testyourlexicon.repositories.AchievementRepository;
+import com.lineate.testyourlexicon.repositories.UserRepository;
+import com.lineate.testyourlexicon.services.AuthenticationService;
+import java.util.ArrayList;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+
+@RequiredArgsConstructor
+public class AchievementManager {
+
+  private final UserRepository userRepository;
+  private final AchievementRepository achievementRepository;
+  private final AuthenticationService authenticationService;
+  private final List<Achievement> achievements = new ArrayList<>();
+
+  public void checkAchievements(Long userHash) {
+    if (!authenticationService.isAuthenticated()) {
+      return;
+    }
```

Comment:
Why do we need to check this? What's the use case when this method can be called and user is not authenticated?